quarkus/config-api/src/main/java/org/keycloak/config/WildcardOptionsUtil.java

```diff
@@ -0,0 +1,121 @@
+     * isExternalWildcardOption("http-port")                    → "false"
+     * isExternalWildcardOption("quarkus.<sth>.end")            → "true"
+     * }</pre>
+     *
+     * @param key the configuration key to check
+     * @return {@code true} if the key represents an external wildcard option
+     */
+    public static boolean isExternalWildcardOption(String key) {
+        return key != null && key.contains(WILDCARD_START) && key.contains(WILDCARD_END);
+    }
+
+    /**
+     * Extracts the prefix part of a wildcard key.
+     * You should always check the presence of the wildcard via the {@link #isWildcardOption(String)}.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * getWildcardPrefix("tracing-header-<header>")       → "tracing-header-"
+     * getWildcardPrefix("db-kind-<datasource>")         → "db-kind-"
+     * }</pre>
+     *
+     * @param wildcardKey a configuration key that includes a wildcard segment
+     * @return the prefix before the wildcard marker, otherwise {@code null}
+     */
+    public static String getWildcardPrefix(String wildcardKey) {
+        return wildcardKey != null ? wildcardKey.substring(0, wildcardKey.indexOf(WILDCARD_START)) : null;
+    }
+
+    /**
+     * Generates a concrete configuration key by replacing the wildcard placeholder with a specific value.
+     * You should always check the presence of the wildcard via the {@link #isWildcardOption(String)}.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * getWildcardNamedKey("tracing-header-<header>", "Authorization")  → "tracing-header-Authorization"
+     * getWildcardNamedKey("db-kind-<datasource>", "user-store") → "db-kind-user-store"
+     * }</pre>
+     *
+     * @param wildcardKey a configuration key that includes a wildcard segment
+     * @param value       the value to replace the wildcard with
+     * @return the resolved key, otherwise {@code null}
+     */
+    public static String getWildcardNamedKey(String wildcardKey, String value) {
+        var prefix = getWildcardPrefix(wildcardKey);
+        return prefix != null ? prefix.concat(value) : null;
+    }
+
+    /**
+     * Extracts the name that replaces the wildcard placeholder from a fully qualified configuration key.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * getWildcardName(TracingOptions.TRACING_HEADER, "tracing-header-Authorization") → "Authorization"
+     * getWildcardName(DatabaseOptions.DB_ENABLE, "db-enable-my-store") → "my-store"
+     * getWildcardName(DatabaseOptions.DB_ENABLE, "kc.db-enable-my-store") → "my-store"
+     * }</pre>
+     *
+     * @param option   the option containing a wildcard key
+     * @param namedKey the fully qualified (resolved) configuration key
+     * @return the part of {@code namedKey} that replaces the wildcard in {@code option.getKey()}, otherwise {@code null}
+     */
+    public static String getWildcardName(Option<?> option, String namedKey) {
```

Comment:
Shouldn't this be rather `getWildcardValue`? It's a value of the `<wildcard>`, not name (which may imply a key), no?